todo-app/src/App.js

```diff
@@ -1,13 +1,83 @@
-import React, { Component } from 'react';
+//jshint esnext: true
+
+import React from 'react';
+
+import TodoList from './Components/todoList';
+import AddTodoItem from './Components/AddTodoItem';
+import './App.css';
+
+
+export default class App extends React.Component {
+
+  
+  state= {
+    todos: [],
+    isEditing: null
+  }
+
+
+  handleToggleEditing = (id) => {
+    this.setState({ isEditing: id || null })
+  }
+
+  handleAddTodo = (text, deadline) => {
+    
+    const newItem = {
+        id: this.state.todos.length + 1 || 1,
+        text,
+        deadline,
+        done: false
+      }
+      const oldTodos = this.state.todos || [];
+      
+      this.setState({ todos: [...oldTodos, newItem]})
+    }
+
+  handleToggleDone = (id) => {
+    const newTodos = [...this.state.todos].map(item => {
```

Comment:
Same as below: map does not mutate the array, so you don't need to `[...<array>]` it here. Just call `this.state.todos.map(...)`!